File: java/SR-RBD/src/ddb/tpc/MessageQueue.java
/**
 * 
 */
package ddb.tpc;

import java.util.concurrent.BlockingQueue;
import java.util.concurrent.LinkedBlockingQueue;

import ddb.Logger;
import ddb.Logger.Level;
import ddb.msg.Message;

/** 
 * <!-- begin-UML-doc -->
 * <!-- end-UML-doc -->
 * @author PKalanski
 */
public class MessageQueue {
	/** 
	 * <!-- begin-UML-doc -->
	 * <!-- end-UML-doc -->
	 */
	//private LinkedList<Message> messages;
	private BlockingQueue<Message> messages;
	
	public MessageQueue() {
		//this.messages = new LinkedList<Message>();
		this.messages = new LinkedBlockingQueue<Message>();
	}
	
	/** 
	 * <!-- begin-UML-doc -->
	 * <!-- end-UML-doc -->
	 * @param message
	 * @throws InterruptedException 
	 */
	public synchronized void putMessage(Message message) throws InterruptedException {
		//TODO:
		messages.put(message);
		//notify();
	}

	/** 
	 * <!-- begin-UML-doc -->
	 * <!-- end-UML-doc -->
	 * @return
	 * @throws InterruptedException 
	 */
	public synchronized Message getMessage() throws InterruptedException {
		//TODO:
		Logger.getInstance().log("Liczba wiadomosci w kolejce: " + messages.size(), "MQ", Level.INFO);
		//return messages.poll();
		return messages.take();
	}
}

File: trunk/C#/RBD/RBD/TPC/MessageQueue.cs
// +--- TODO BlockingQueue

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using RBD.Msg;
using RBD.Util;

namespace RBD.TPC
{
    public class MessageQueue
    {
        private BlockingQueue<Message> messages = new BlockingQueue<Message>();

        /** 
         * <!-- begin-UML-doc -->
         * <!-- end-UML-doc -->
         * @param message
         * @throws InterruptedException 
         */
        public void putMessage(Message message) //throws InterruptedException
        {
            messages.put(message);
        }

        /** 
         * <!-- begin-UML-doc -->
         * <!-- end-UML-doc -->
         * @return
         * @throws InterruptedException 
         */
        public Message getMessage() //throws InterruptedException 
        {
            //return new HelloMessage();
            return messages.take();
        }
    }
}
